internal/controller/postgrescluster/instance.go

```diff
@@ -1201,16 +1201,30 @@ func (r *Reconciler) reconcileInstance(
 	}
 
 	if err == nil &&
-		(feature.Enabled(ctx, feature.OpenTelemetryLogs) || feature.Enabled(ctx, feature.OpenTelemetryMetrics)) {
+		(feature.Enabled(ctx, feature.OpenTelemetryLogs) && !feature.Enabled(ctx, feature.OpenTelemetryMetrics)) {
+
 		// TODO: Setting the includeLogrotate argument to false for now. This
 		// should be changed when we implement log rotation for postgres
 		collector.AddToPod(ctx, cluster.Spec.Instrumentation, cluster.Spec.ImagePullPolicy, instanceConfigMap, &instance.Spec.Template.Spec,
 			[]corev1.VolumeMount{postgres.DataVolumeMount()}, "", false)
 	}
 
-	// Add pgMonitor resources to the instance Pod spec
-	if err == nil {
-		err = addPGMonitorToInstancePodSpec(ctx, cluster, &instance.Spec.Template, exporterQueriesConfig, exporterWebConfig)
+	if err == nil &&
+		feature.Enabled(ctx, feature.OpenTelemetryMetrics) {
+
+		monitoringUserSecret := &corev1.Secret{ObjectMeta: naming.MonitoringUserSecret(cluster)}
+		err = errors.WithStack(
+			r.Client.Get(ctx, client.ObjectKeyFromObject(monitoringUserSecret), monitoringUserSecret))
```

Comment:
As of right now, the monitoring user secret will only exist if exporter is enabled. This should be adjusted in the `reconcileMonitoringSecret` method in the `controller/postgrescluster/pgmonitor.go` file.

Reply:
That would be a cleaner design. Creation of the monitoring user and related setup is driven by exporter being enabled. I'll check on what it takes to refactor that piece.

Reply:
We're now configured so that you can run the collector off the feature gate alone, and the feature gate takes precedence over the exporter spec.